### PR TITLE
Add authentication (FE)

### PR DESCRIPTION
**Pull request by CahidArda**
I have completed implementation of the authentication & token handling.

## Sign-in / Log-out

- If you click the profile avatar to the right of the navigation bar in any page, you will see options to sign-in/sign-up or log-out:

image

image

Menu items are all functional and work as you would expect. If you go to sign in and enter e-mail and password (any e-mail & password until backend gives us an endpoint), you will be directed to the home page. When you click on the menu now, you will be able to log-out. If you log-out, you will be in the same state you were in before logging in.

## Protected Pages

You can now protect pages like the create art item page by using the `ProtectedRoute` component:
```js
<Route
  path="/new/art_item"
  element={<ProtectedRoute><CreateArtItemPage /></ProtectedRoute>} 
/>
```

## Updating Components

Now that we can check whether a token exists, I have updated some components.

### Navigation Bar

Change is as explained in the Sign-in / Log-out section I wrote above.

### User Cards

Follow user button is hidden.

image

### Comments Section

Like comment buttons and write new comment sections are hidden.

image

### CODE REVIEW

**Comment by erimerkin** [comment]
While testing the production build with this PR, I encountered some error. The error's source is not the additions, I will fix this error and push the changes to this branch. So the error is fixed in both this and main frontend branch once this is merged. Is it OK for you, @CahidArda?


**Error Message:**
```
=> [internal] load build definition from Dockerfile.production            0.0s
 => => transferring dockerfile: 48B                                        0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 35B                                           0.0s
 => [internal] load metadata for docker.io/library/nginx:1.22.1-alpine     1.5s
 => [internal] load metadata for docker.io/library/node:16-alpine          1.5s
 => [builder 1/5] FROM docker.io/library/node:16-alpine@sha256:2175727cef  0.0s
 => [internal] load build context                                          0.2s
 => => transferring context: 12.34MB                                       0.2s
 => [production 1/3] FROM docker.io/library/nginx:1.22.1-alpine@sha256:5b  0.0s
 => CACHED [builder 2/5] WORKDIR /app                                      0.0s
 => [builder 3/5] COPY . .                                                 0.1s
 => ERROR [builder 4/5] RUN npm i                                         29.5s
------                                                                          
 > [builder 4/5] RUN npm i:                                                     
#10 7.277 npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility                                    
#10 29.21 npm ERR! code 1
#10 29.21 npm ERR! path /app/node_modules/lmdb
#10 29.21 npm ERR! command failed
#10 29.21 npm ERR! command sh -c -- node-gyp-build-optional-packages
#10 29.21 npm ERR! gyp info it worked if it ends with ok
#10 29.21 npm ERR! gyp info using node-gyp@9.1.0
#10 29.21 npm ERR! gyp info using node@16.18.0 | linux | x64
#10 29.22 npm ERR! gyp ERR! find Python 
#10 29.22 npm ERR! gyp ERR! find Python Python is not set from command line or npm configuration
#10 29.22 npm ERR! gyp ERR! find Python Python is not set from environment variable PYTHON
#10 29.22 npm ERR! gyp ERR! find Python checking if "python3" can be used
#10 29.22 npm ERR! gyp ERR! find Python - "python3" is not in PATH or produced an error
#10 29.22 npm ERR! gyp ERR! find Python checking if "python" can be used
#10 29.22 npm ERR! gyp ERR! find Python - "python" is not in PATH or produced an error
#10 29.22 npm ERR! gyp ERR! find Python 
#10 29.22 npm ERR! gyp ERR! find Python **********************************************************
#10 29.22 npm ERR! gyp ERR! find Python You need to install the latest version of Python.
#10 29.22 npm ERR! gyp ERR! find Python Node-gyp should be able to find and use Python. If not,
#10 29.22 npm ERR! gyp ERR! find Python you can try one of the following options:
#10 29.22 npm ERR! gyp ERR! find Python - Use the switch --python="/path/to/pythonexecutable"
#10 29.22 npm ERR! gyp ERR! find Python   (accepted by both node-gyp and npm)
#10 29.22 npm ERR! gyp ERR! find Python - Set the environment variable PYTHON
#10 29.22 npm ERR! gyp ERR! find Python - Set the npm configuration variable python:
#10 29.22 npm ERR! gyp ERR! find Python   npm config set python "/path/to/pythonexecutable"
#10 29.22 npm ERR! gyp ERR! find Python For more information consult the documentation at:
#10 29.22 npm ERR! gyp ERR! find Python https://github.com/nodejs/node-gyp#installation
#10 29.22 npm ERR! gyp ERR! find Python **********************************************************
#10 29.22 npm ERR! gyp ERR! find Python 
#10 29.22 npm ERR! gyp ERR! configure error 
#10 29.22 npm ERR! gyp ERR! stack Error: Could not find any Python installation to use
#10 29.22 npm ERR! gyp ERR! stack     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:330:47)
#10 29.22 npm ERR! gyp ERR! stack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:159:21)
#10 29.22 npm ERR! gyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:202:16)
#10 29.22 npm ERR! gyp ERR! stack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:294:16)
#10 29.22 npm ERR! gyp ERR! stack     at exithandler (node:child_process:410:5)
#10 29.23 npm ERR! gyp ERR! stack     at ChildProcess.errorhandler (node:child_process:422:5)
#10 29.23 npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
#10 29.23 npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
#10 29.23 npm ERR! gyp ERR! stack     at onErrorNT (node:internal/child_process:485:16)
#10 29.23 npm ERR! gyp ERR! stack     at processTicksAndRejections (node:internal/process/task_queues:83:21)
#10 29.23 npm ERR! gyp ERR! System Linux 5.10.104-linuxkit
#10 29.23 npm ERR! gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
#10 29.23 npm ERR! gyp ERR! cwd /app/node_modules/lmdb
#10 29.23 npm ERR! gyp ERR! node -v v16.18.0
#10 29.23 npm ERR! gyp ERR! node-gyp -v v9.1.0
#10 29.23 npm ERR! gyp ERR! not ok
#10 29.23 
#10 29.23 npm ERR! A complete log of this run can be found in:
#10 29.23 npm ERR!     /root/.npm/_logs/2022-10-25T07_55_10_947Z-debug-0.log
------
executor failed running [/bin/sh -c npm i]: exit code: 1```

**Comment by CahidArda** [comment]
That's okay by me.

File: frontend/src/pages/Authentication/SignInPage.js [hunk 1]
```diff
@@ -21,14 +22,16 @@ function SignInForm(props) {
...
 
   // Called when the user clicks the submit button. Observe how
   // the handleSubmit button is attached to the 'form' component
   // below in the return call of SignUpForm function.
   const handleSubmit = event => {
     event.preventDefault();
-
-    let data = { formInput };
+    
+    console.log(formInput)
```
Comment (erimerkin) [inline frontend/src/pages/Authentication/SignInPage.js:33]: Is this necessary? If not should be removed.

**Review by erimerkin — COMMENTED** [review]
Looks good @CahidArda, my only concern is `console.log()` part.  I think we should delete them before merging.

File: frontend/src/pages/Authentication/SignInPage.js (hunk 1 above)
Comment (CahidArda) [reply frontend/src/pages/Authentication/SignInPage.js]: It is not necessary. I forgot it there after testing.

**Review by erimerkin — APPROVED** [review]
Everything looks good, we can merge this PR.

**Pull request merged**